### PR TITLE
fix: Correcting namespace for csi-secret-store-provider-aws module

### PR DESCRIPTION
**Pull request by MaiCabrera**
What does this PR do?
🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Motivation
Trying to just leverage the csi secret store module to mount secrets to my main microservices

Resolves https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1172

### CODE REVIEW

File: modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 locals {
   name      = try(var.helm_config.name, "csi-secrets-store-provider-aws")
-  namespace = try(var.helm_config.namespace, local.name)
+  namespace = "kube-system"
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf:3]: Looks correct https://github.com/aws/secrets-store-csi-driver-provider-aws/blob/41dc61e8518568bf371c1f22fe8b272ff998736a/deployment/aws-provider-installer.yaml#L6

**Review by bryantbiggs — APPROVED** [review]

File: modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf (hunk 1 above)
Comment (bryantbiggs) [reply modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf:3]:
```suggestion
  namespace = try(var.helm_config.namespace, "kube-system")
```

**Review by bryantbiggs — APPROVED** [review]

**Comment by nahum-landa** [comment]
Error: namespaces "kube-system" already exists
│ 
│   with module.kubernetes_addons.module.csi_secrets_store_provider_aws[0].kubernetes_namespace_v1.csi_secrets_store_provider_aws,
│   on .terraform/modules/kubernetes_addons/modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf line 6, in resource "kubernetes_namespace_v1" "csi_secrets_store_provider_aws":
│    6: resource "kubernetes_namespace_v1" "csi_secrets_store_provider_aws" {

**Comment by csantanapr** [comment]
Creating the namespace should be optional now
https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf#L6

**Comment by lordz-ei** [comment]
> Creating the namespace should be optional now https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/csi-secrets-store-provider-aws/main.tf#L6

Nope, still tries to create namespace if namespace is kube-system.

**Comment by lordz-md** [comment]
I believe this is partially fixed here [https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/main/modules/csi-secrets-store-provider-aws/main.tf#L12-L18](url)
but I think code should be 
`resource "kubernetes_namespace_v1" "csi_secrets_store_provider_aws" {
  metadata {
    name = local.namespace

    # Only create the namespace if the name is not kube-system
    # If the name is kube-system, this block will be skipped
    count = local.namespace != "kube-system" ? 1 : 0
  }
}`

**Comment by saibossai** [comment]
Created the pr: #1525 to fix the namespace creation issue if the namespace: `kube-system` already exists.

**Pull request merged**